tests/fgpyo/sam/test_cigar.py

```diff
@@ -31,3 +31,51 @@ def test_bad_index_raises_index_error(index: int) -> None:
 def test_bad_index_raises_type_error(index: int) -> None:
     with pytest.raises(TypeError):
         cigar[index]
+
+
+@pytest.mark.parametrize(
+    ("cigar_string", "start", "end"),
+    {
+        ("10M", 0, 10),
+        ("10M10I", 0, 20),
+        ("10X10I", 0, 20),
+        ("10X10D", 0, 10),
+        ("10=10D", 0, 10),
+        ("10S10M", 10, 20),
+        ("10H10M", 0, 10),
+        ("10H10S10M", 10, 20),
+        ("10H10S10M5S", 10, 20),
+        ("10H10S10M5S10H", 10, 20),
+        ("10H", 0, 0),
+        ("10S", 10, 10),
+        ("10S10H", 10, 10),
+        ("5H10S10H", 10, 10),
```

Comment:
**issue** none of these CIGAR strings have aligned sequence, so the variability in the start and end offsets is confusing to me.

(See also my question above - why are the start and end 0 in some cases, the length of the CIGAR in others, and somewhere in the middle in still others?)

Reply:
explained in the updated docs

Reply:
I do not think any of these should return values, as noted above.